### ISSUE TITLE
Don't lint code in pragmas such as `RULE`

### ISSUE THREAD

**Issue by sgraf812** [ndmitchell/hlint#374]
I have a rule for fusing `map`s together, like so:

```
{-# RULES
"map/map" forall f g xs . map f (map g xs) = map (f . g) xs
 #-}
```

`hlint` recommends that I should replace the LHS by the RHS of that equation, which is an amusing false positive.

**Comment by ndmitchell** [comment]
LOL! A genius hint - very much unexpected consequences.

**Comment by ndmitchell** [comment]
Done as part of #633
